framework/decode/custom_vulkan_struct_decoders.cpp

```diff
@@ -110,6 +110,45 @@ size_t DecodeStruct(const uint8_t* buffer, size_t buffer_size, Decoded_VkAcceler
     return 0;
 }
 
+size_t DecodeStruct(const uint8_t* buffer, size_t buffer_size, Decoded_VkAccelerationStructureMotionInstanceNV* wrapper)
+{
+    assert((wrapper != nullptr) && (wrapper->decoded_value != nullptr));
+
+    size_t                                   bytes_read = 0;
+    VkAccelerationStructureMotionInstanceNV* value      = wrapper->decoded_value;
+
+    bytes_read += ValueDecoder::DecodeEnumValue((buffer + bytes_read), (buffer_size - bytes_read), &(value->type));
+    bytes_read += ValueDecoder::DecodeFlagsValue((buffer + bytes_read), (buffer_size - bytes_read), &(value->flags));
+
+    switch (value->type)
+    {
+        case VK_ACCELERATION_STRUCTURE_MOTION_INSTANCE_TYPE_STATIC_NV:
+            wrapper->staticInstance = DecodeAllocator::Allocate<Decoded_VkAccelerationStructureInstanceKHR>();
+            wrapper->staticInstance->decoded_value =
+                reinterpret_cast<VkAccelerationStructureInstanceKHR*>(&value->data);
```

Comment:
You can skip the cast by reaching further into `data`, ie:
```
            wrapper->staticInstance->decoded_value = &value->data.staticInstance;
```
Totally up to you, it shouldn't change the behavior

Reply:
Thanks. You're right.